sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/object.scala

```diff
@@ -41,7 +41,20 @@ object CatalystSerde {
   }
 
   def generateObjAttr[T : Encoder]: Attribute = {
-    AttributeReference("obj", encoderFor[T].deserializer.dataType, nullable = false)()
+    val deserializer = encoderFor[T].deserializer
+    val dataType = deserializer.dataType
+    val nullable = if (deserializer.childrenResolved) {
```

Comment:
Good point. When this method is executed in other call context, `deserializer` has been already resolved. The following is one of examples:

```java
	at org.apache.spark.sql.catalyst.plans.logical.CatalystSerde$.generateObjAttr(object.scala:47)
	at org.apache.spark.sql.catalyst.plans.logical.CatalystSerde$.deserialize(object.scala:36)
	at org.apache.spark.sql.catalyst.encoders.ExpressionEncoder.resolveAndBind(ExpressionEncoder.scala:254)
	at org.apache.spark.sql.Dataset.<init>(Dataset.scala:207)
	at org.apache.spark.sql.Dataset$.ofRows(Dataset.scala:68)
	at org.apache.spark.sql.Dataset.org$apache$spark$sql$Dataset$$withPlan(Dataset.scala:2831)
	at org.apache.spark.sql.Dataset.createOrReplaceTempView(Dataset.scala:2612)
	at org.apache.spark.sql.test.SQLTestData$class.testData(SQLTestData.scala:54)
	at org.apache.spark.sql.DataFrameSuite.testData$lzycompute(DataFrameSuite.scala:43)
	at org.apache.spark.sql.DataFrameSuite.testData(DataFrameSuite.scala:43)
...
```